fluss-client/src/main/java/org/apache/fluss/client/metadata/MetadataUpdater.java

```diff
@@ -293,8 +296,24 @@ private static Cluster initializeCluster(Configuration conf, RpcClient rpcClient
         Exception lastException = null;
         for (InetSocketAddress address : inetSocketAddresses) {
             try {
-                cluster = tryToInitializeCluster(rpcClient, address);
-                break;
+                ServerNode serverNode =
+                        new ServerNode(
+                                -1,
+                                address.getHostString(),
+                                address.getPort(),
+                                ServerType.COORDINATOR);
+                AdminReadOnlyGateway adminReadOnlyGateway =
+                        GatewayClientProxy.createGatewayProxy(
+                                () -> serverNode, rpcClient, AdminReadOnlyGateway.class);
+                if (inetSocketAddresses.size() == 1) {
+                    // if there is only one bootstrap server, we can retry to connect to it.
+                    cluster =
+                            tryToInitializeClusterWithRetries(
+                                    rpcClient, serverNode, adminReadOnlyGateway, MAX_RETRY_TIMES);
+                } else {
+                    cluster = tryToInitializeCluster(adminReadOnlyGateway);
+                    break;
+                }
             } catch (Exception e) {
                 LOG.error(
                         "Failed to initialize fluss client connection to bootstrap server: {}",
```

Comment:
We should improve this exception message, it didn't carry exception message to the top error message which may confuse users. 

```
            String errorMsg =
                    "Failed to initialize fluss client connection to bootstrap servers: "
                            + inetSocketAddresses
                            + ". \nReason: "
                            + lastException.getMessage();
```